plugins/kafka/kafka.cpp

```diff
@@ -0,0 +1,1118 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "kafka.hpp"
+
+#include "rtlds_imp.hpp"
+#include "jlog.hpp"
+#include "jmutex.hpp"
+#include "jprop.hpp"
+#include "jfile.hpp"
+#include "build-config.h"
+
+#include "librdkafka/rdkafka.h"
+
+#include <map>
+#include <fstream>
+
+//==============================================================================
+// Kafka Interface Code
+//==============================================================================
+
+namespace KafkaPlugin
+{
+    //--------------------------------------------------------------------------
+    // File Constants
+    //--------------------------------------------------------------------------
+
+    // The minimum number of seconds that a cached object can live
+    // without activity
+    const time_t    OBJECT_EXPIRE_TIMEOUT_SECONDS = 60 * 2;
+
+    // The number of milliseconds given to librdkafka to perform explicit
+    // background activity
+    const __int32   POLL_TIMEOUT = 1000;
+
+    //--------------------------------------------------------------------------
+    // Static Variables
+    //--------------------------------------------------------------------------
+
+    static CriticalSection    initLock;
+
+    //--------------------------------------------------------------------------
+    // Static Methods (internal)
+    //--------------------------------------------------------------------------
+
+    /**
+     * Look for an optional configuration file and apply any found configuration
+     * parameters to a librdkafka configuration object.
+     *
+     * @param   configFilePath      The path to a configuration file; it is not
+     *                              necessary for the file to exist
+     * @param   globalConfigPtr     A pointer to the configuration object that
+     *                              will receive any found parameters
+     * @param   traceLevel          The current log trace level
+     */
+    static void applyConfig(const char* configFilePath, RdKafka::Conf* globalConfigPtr, __int32 traceLevel)
+    {
+        if (configFilePath && *configFilePath && globalConfigPtr)
+        {
+            std::string                 errStr;
```

Comment:
minor:  We tend to not align variable names.  Generally a single space between the type and the name.  Less of an issue in a completely new independent source file.